tower-sessions-core/src/session.rs

```diff
@@ -856,27 +855,38 @@ impl Session {
 ///
 /// Id::default();
 /// ```
-#[derive(Copy, Clone, Debug, Deserialize, Serialize, Eq, Hash, PartialEq)]
-pub struct Id(pub Uuid); // TODO: By this being public, it may be possible to override UUIDv4,
-                         // which is undesirable.
+#[derive(Copy, Clone, Debug, Deserialize, Eq, Hash, PartialEq)]
+pub struct Id(pub [u8; 22]); // TODO: By this being public, it may be possible to override the
+                             // session ID, which is undesirable.
 
 impl Default for Id {
     fn default() -> Self {
-        Self(Uuid::new_v4())
+        let id = nanoid::nanoid!(22);
+        Self(id.as_bytes().try_into().unwrap())
     }
 }
 
 impl Display for Id {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str(&self.0.as_hyphenated().to_string())
+        let data = std::str::from_utf8(&self.0).unwrap();
```

Comment:
We probably want to ensure we don't introduce a potential panic here.

Reply:
Fixed in the commit I just pushed; Instead of using a byte array, I'm just using an i128; That makes it relatively easy to use the base64 crate over the bytes in little-endian format.